File: AOJ/ITP1/itp_1_5_b.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	var h, w int
	for {
		fmt.Scan(&h, &w)
		if h == 0 && w == 0 {
			return
		}
		fmt.Println(strings.Repeat("#", w))
		for i := 0; i < h-2; i++ {
			fmt.Printf("#%v#\n", strings.Repeat(".", w-2))
		}
		fmt.Println(strings.Repeat("#", w))
		fmt.Println()
	}
}

#include <bits/stdc++.h>
using namespace std;
int main() {
  int n;
  int a[n];
  cin >> n;
  for (int i=0;i<n;i++){
    cin >> a[i];
  }
  for (int i=n-1;i>=0;i--){
    if (i != n-1){
      cout << " ";
    }
    cout << a[i];
  }
  cout << endl;
}

File: AOJ/ITP1/itp_1_5_b.cpp
#include <bits/stdc++.h>
using namespace std;
int main() {
  int h, w;
  while (true) {
    scanf("%d %d", &h, &w);
    if ((h == 0) && (w == 0)) break;
    for (int i = 0; i < h; i++) {
      for (int j = 0; j < w; j++) {
        if ((i == 0) || (i == h - 1)) {
          printf("#");
          continue;
        }
        if ((j == 0) || (j == w - 1)) {
          printf("#");
          continue;
        }
        printf(".");
      }
      printf("\n");
    }
    printf("\n");
  }
}
